### DISCUSSION TITLE
Auth0 + Rest API Datasource + CRUD

### DISCUSSION THREAD

**Discussion by fmontada** [Budibase/budibase#7920] [Help]
Hi team, 

We want to use Budibase to build a few of the Internal admin tools we need, and we have a few questions 

1. Do you have a Template/Doc we can use to create a CRUD App using Rest API Datasource? 
3. We are using Auth0 as our Idp. do we have to set the Authorization Header, or do we have to use the Authentication UI?  
![image](https://user-images.githubusercontent.com/9787595/191835874-05f1abf4-f953-4353-b0b2-696e05f32e46.png)
4. Can we use dynamic values for the token 
![image](https://user-images.githubusercontent.com/9787595/191836322-059a6685-3447-4735-999a-f9f0bc1b5f97.png)

**Comment by melohagan** [comment]
Hey @fmontada 

I don't think there's an app template using the REST API, however it is fairly well documented: https://docs.budibase.com/docs/rest

As you are using Auth0 to log-in, you should have access to the [OAuth token](https://docs.budibase.com/docs/authentication-and-sso#using-the-oauth-token) which you can add to the REST datasource headers as needed.
